packages/contracts-periphery/contracts/universal/DrippieRedd.sol

```diff
@@ -0,0 +1,152 @@
+ * @notice DrippieRedd goes brrr.
+ */
+contract DrippieRedd is RetroReceiver {
+    /**
+     * Enum representing different status options for a given drip.
+     */
+    enum DripStatus {
+        NONE,
+        ACTIVE,
+        PAUSED
+    }
+
+    /**
+     * Represents the configuration for a given drip.
+     */
+    struct DripConfig {
+        address payable recipient;
+        bytes data;
+        bytes checkscript;
+        uint256 amount;
+        uint256 interval;
+    }
+
+    /**
+     * Represents the state of an active drip.
+     */
+    struct DripState {
+        DripStatus status;
+        DripConfig config;
+        uint256 last;
+    }
+
+    /**
+     * Maps from drip names to drip configurations.
+     */
+    mapping(string => DripState) public drips;
+
+    /**
+     * @param _owner Initial owner address.
+     */
+    constructor(address _owner) RetroReceiver(_owner) {}
+
+    /**
+     * Creates a new drip with the given name and configuration.
+     *
+     * @param _name Name of the drip.
+     * @param _config Configuration for the drip.
+     */
+    function create(string memory _name, DripConfig memory _config) public onlyOwner {
+        require(
+            drips[_name].status == DripStatus.NONE,
+            "DrippieRedd: drip with that name already exists"
+        );
+
+        drips[_name] = DripState({ status: DripStatus.ACTIVE, config: _config, last: 0 });
+    }
+
+    /**
+     * Configures a drip by name.
+     *
+     * @param _name Name of the drip to configure.
+     * @param _config Drip configuration.
+     */
+    function update(string memory _name, DripConfig memory _config) public onlyOwner {
+        require(
+            drips[_name].status != DripStatus.NONE,
+            "DrippieRedd: drip with that name does not exist"
+        );
+
+        drips[_name].config = _config;
+    }
+
+    /**
+     * Toggles the status of a given drip.
+     *
+     * @param _name Name of the drip to toggle.
+     */
+    function toggle(string memory _name) public onlyOwner {
+        require(
+            drips[_name].status != DripStatus.NONE,
+            "DrippieRedd: drip with that name does not exist"
+        );
+
+        if (drips[_name].status == DripStatus.ACTIVE) {
+            drips[_name].status = DripStatus.PAUSED;
+        } else {
+            drips[_name].status = DripStatus.ACTIVE;
+        }
+    }
+
+    /**
+     * Triggers a drip.
+     *
+     * @param _name Name of the drip to trigger.
+     */
+    function drip(string memory _name) public {
+        DripState storage state = drips[_name];
+
+        require(
+            state.status == DripStatus.ACTIVE,
+            "DrippieRedd: selected drip does not exist or is not currently active"
+        );
+
+        require(
+            state.last + state.config.interval <= block.timestamp,
+            "DrippieRedd: drip interval has not elapsed"
+        );
+
+        bytes memory checkscript = abi.encodePacked(
```

Comment:
iiuc, this is just a way to get the output of a script effecting anything? What happens if the user script reverts? Or doesn't expect the recipient address?

Reply:
Heavily documented this in comments now